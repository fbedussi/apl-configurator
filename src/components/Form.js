import React from 'react';
import { connect } from 'react-redux';

import { getShowForm } from '../selectors';
import { requestSent } from '../actions';

const mapStateToProps = (state) => ({
    showForm: getShowForm(state)
});

const mapDispatchToProps = (dispatch) => ({
    requestSent: () => dispatch(requestSent())
})

class Form extends React.Component {
    handleSubmit(e) {
        console.log(e);
        var form = e.target;
        var data = new FormData(form);
        fetch('send_form.php', {
            method: 'post',
            body: data
        })
        .then(response => {
            console.log(response);
            this.props.requestSent();
        })
        .catch(err => {
            console.log(err);
        })
    }

    render() {
        return <form onSubmit={(e) => this.handleSubmit(e)}
            className={`quotation-form ${this.props.showForm? 'show' : 'hide' }`}
            name="contactForm"
        >
            <label className="quotation-field" htmlFor="name">
                Nome
            </label>
                <input type="text" id="name" name="name"/>
            <label className="quotation-field">
                Cognome
                <input type="text" id="surname" name="surname"/>
            </label>
            <label className="quotation-field">
                Societ√†* (ragione sociale)
                <input type="text" id="company" name="company"/>
            </label>
            <label className="quotation-field">
                Indirizzo*
                <input type="text" id="address" name="address"/>
            </label>
            <label className="quotation-field">
                Posizione
                <input type="text" id="role" name="role"/>
            </label>
            <label className="quotation-field">
                e-mail*
                <input type="email" id="email" name="email"/>
            </label>
            <label className="quotation-field">
                Telefono
                <input type="tel" id="tel" name="tel"/>
            </label>
            <label className="quotation-field">
                Sito
                <input type="text" id="website" name="website"/>
            </label>
            <fieldset className="quotation-field">
                Motivo della richiesta:
                <label>
                    <input type="radio" name="reason" id="peoject"/>
                    Elaborazione progetto
                </label>
                <label>
                    <input type="radio" name="reason" id="call"/>
                    Miglioria per gara d'appalto
                </label>
                <label>
                    <input type="radio" name="reason" id="other"/>
                    Altro
                </label>
            </fieldset>
            <label className="quotation-field">
                Eventuali informazioni aggiuntive:
                <textarea className="quotation-msg" id="msg" name="msg"></textarea>
            </label>

            <button className="ctaBtn" type="submit">Invia</button>
        </form>
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Form);
