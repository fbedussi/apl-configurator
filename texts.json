{
    "it": {
        "id": 1,
        "type": "question",
        "text": "Nel sito di installazione è possibile collegare ol sistema APL alla rete elettrica?",
        "answers": [
            {
                "a": "no"
            },
            {
                "b": "sì"
            }
        ],
        "a": {
            "id": 2,
            "type": "answer",
            "title": "Apl solar",
            "text": "Questa soluzione è valida solo se nel sito di installazione non è presente illuminazione pubblica in quanto le potenze sviluppate non sono sufficienti a creare un contrasto adeguato se non in zone buie"
        },
        "b": {
            "id": 3,
            "type": "question",
            "text": "Il sistema deve essere dotato di dispositivo per l'interazione con il pedone?",
            "answers": [
                {
                    "a": "no"
                },
                {
                    "b": "sì"
                }
            ],
            "a": {
                "id": 4,
                "type": "question",
                "text": "Nel luogo di installazione è già presente illuminazione stradale oppure altra illuminazione artificiale? (Vetrine, insegne luminose, ecc.)?",
                "answers": [
                    {
                        "a": "no"
                    },
                    {
                        "b": "sì"
                    }
                ],
                "a": {
                    "id": 5,
                    "type": "question",
                    "text": "La distanza tra i 2 pali è maggiore di 9 metri?",
                    "answers": [
                        {
                            "a": "no"
                        },
                        {
                            "b": "sì"
                        }
                    ],
                    "a": {
                        "id": 6,
                        "type": "question",
                        "text": "Ci sono ostacoli dove verranno installati i pali che potrebbero ridurre la visibilità dei cartelli di segnalazione laterali (alberi, tabelle segnaletiche o pubblicitarie, ecc)?",
                        "answers": [
                            {
                                "a": "no"
                            },
                            {
                                "b": "sì"
                            }
                        ],
                        "a": {
                            "id": 7,
                            "type": "answer",
                            "title": "Apl classic Stratos N soluzione 1",
                            "text": "Testo"
                        },
                        "b": {
                            "id": 8,
                            "type": "answer",
                            "title": "Apl classic Stratos N soluzione 2",
                            "text": "Testo"
                        }
                    },
                    "b": {
                        "id": 9,
                        "type": "question",
                        "text": "Ci sono ostacoli dove verranno installati i pali che potrebbero ridurre la visibilità dei cartelli di segnalazione laterali (alberi, tabelle segnaletiche o pubblicitarie, ecc)?",
                        "answers": [
                            {
                                "a": "no"
                            },
                            {
                                "b": "sì"
                            }
                        ],
                        "a": {
                            "id": 10,
                            "type": "answer",
                            "title": "Apl classic Stratos P soluzione 1",
                            "text": "Testo"
                        },
                        "b": {
                            "id": 11,
                            "type": "answer",
                            "title": "Apl classic Stratos P soluzione 2",
                            "text": "Testo"
                        }
                    }
                },
                "b": {
                    "id": 12,
                    "type": "question",
                    "text": "Ci sono ostacoli dove verranno installati i pali che potrebbero ridurre la visibilità dei cartelli di segnalazione laterali (alberi, tabelle segnaletiche o pubblicitarie, ecc)?",
                    "answers": [
                        {
                            "a": "no"
                        },
                        {
                            "b": "sì"
                        }
                    ],
                    "a": {
                        "id": 13,
                        "type": "answer",
                        "title": "Apl classic Stratos P soluzione 1",
                        "text": "Testo"
                    },
                    "b": {
                        "id": 14,
                        "type": "answer",
                        "title": "Apl classic Stratos P soluzione 2 o 3",
                        "text": "Testo"
                    }
                }
            },
            "b": {
                "id": 15,
                "type": "question",
                "text": "L'interazione con il pedone deve avvenire tramite pulsante touch (consigliata) o tramite sensore? Scarica le istruzioni posizionamento sensore per valutare se è possibile utilizzarlo nella tua installazione",
                "answers": [
                    {
                        "a": "Sensore"
                    },
                    {
                        "b": "Pulsante touch"
                    }
                ],
                "a": {
                    "id": 16,
                    "type": "question",
                    "text": "Il limite di velocità consentito sulla strada è maggiore di 50km/h?",
                    "answers": [
                        {
                            "a": "no"
                        },
                        {
                            "b": "sì"
                        }
                    ],
                    "a": {
                        "id": 17,
                        "type": "question",
                        "text": "Ci sono ostacoli dove verranno installati i pali che potrebbero ridurre la visibilità dei cartelli di segnalazione laterali (alberi, tabelle segnaletiche o pubblicitarie, ecc)?",
                        "answers": [
                            {
                                "a": "no"
                            },
                            {
                                "b": "sì"
                            }
                        ],
                        "a": {
                            "id": 18,
                            "type": "answer",
                            "title": "Apl smart soluzione 1 con sensore",
                            "text": "Testo"
                        },
                        "b": {
                            "id": 19,
                            "type": "answer",
                            "title": "Apl smart soluzione 2 o 3 con sensore",
                            "text": "Testo"
                        }
                    },
                    "b": {
                        "id": 20,
                        "type": "answer",
                        "title": "Apl smart soluzione 2 o 3 con sensore",
                        "text": "Testo"
                    }
                },
                "b": {
                    "id": 21,
                    "type": "question",
                    "text": "Il limite di velocità consentito sulla strada è maggiore di 50km/h?",
                    "answers": [
                        {
                            "a": "no"
                        },
                        {
                            "b": "sì"
                        }
                    ],
                    "a": {
                        "id": 22,
                        "type": "question",
                        "text": "Ci sono ostacoli dove verranno installati i pali che potrebbero ridurre la visibilità dei cartelli di segnalazione laterali (alberi, tabelle segnaletiche o pubblicitarie, ecc)?",
                        "answers": [
                            {
                                "a": "no"
                            },
                            {
                                "b": "sì"
                            }
                        ],
                        "a": {
                            "id": 23,
                            "type": "answer",
                            "title": "Apl smart soluzione 1 con pulsante touch",
                            "text": "Testo"
                        },
                        "b": {
                            "id": 24,
                            "type": "answer",
                            "title": "Apl smart soluzione 2 o 3 con pulsante touch",
                            "text": "Testo"
                        }
                    },
                    "b": {
                        "id": 25,
                        "type": "answer",
                        "title": "Apl smart soluzione 2 o 3 con pulsante touch",
                        "text": "Testo"
                    }
                }
            }
        }
    }
}