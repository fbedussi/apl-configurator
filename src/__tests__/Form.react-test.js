import React from 'react';
import Form from '../components/Form';
import renderer from 'react-test-renderer';
import texts from '../fakeData/apl-texts.json'
import {shallow} from 'enzyme';   

const submitHandler = jest.fn();

const props = {
  labels: texts.labels,
  answer: texts.answers[0],
  submitHandler
}

test('Form component', () => {
  const component = renderer.create(
    <Form {...props} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Form component - call handler when submit', () => {
  const component = shallow(
    <Form {...props} />
  );

  component.find({name: 'company'}).simulate('change', {target: {value: 'test'}});
  component.find({name: 'address'}).simulate('change', {target: {value: 'test'}});
  component.find({name: 'email'}).simulate('change', {target: {value: 'test@test.com'}});
  component.find({id: 'project'}).simulate('change', {target: {checked: true}});
  component.simulate('submit');

  expect(submitHandler).toBeCalled();
});