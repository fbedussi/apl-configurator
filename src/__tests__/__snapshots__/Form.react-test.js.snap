// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component 1`] = `
<form
  className="quotation-form"
  name="contactForm"
  onSubmit={[Function]}
>
  <p
    className="form-mandatory"
  >
    * campi obbligatori
  </p>
  <input
    id="solution"
    name="solution"
    type="hidden"
    value="APL Solar-active - con pulsante touch"
  />
  <div
    className="form-field"
  >
    <label
      className="form-label"
      htmlFor="name"
    >
      Nome
    </label>
    <input
      className="form-input"
      id="name"
      minLength={undefined}
      name="name"
      required={false}
      type="text"
    />
  </div>
  <div
    className="form-field"
  >
    <label
      className="form-label"
      htmlFor="surname"
    >
      Cognome
    </label>
    <input
      className="form-input"
      id="surname"
      minLength={undefined}
      name="surname"
      required={false}
      type="text"
    />
  </div>
  <div
    className="form-field"
  >
    <label
      className="form-label"
      htmlFor="company"
    >
      Societ√† (ragione sociale)*
    </label>
    <input
      className="form-input"
      id="company"
      minLength={undefined}
      name="company"
      required={true}
      type="text"
    />
  </div>
  <div
    className="form-field"
  >
    <label
      className="form-label"
      htmlFor="address"
    >
      Indirizzo*
    </label>
    <input
      className="form-input"
      id="address"
      minLength={undefined}
      name="address"
      required={true}
      type="text"
    />
  </div>
  <div
    className="form-field"
  >
    <label
      className="form-label"
      htmlFor="email"
    >
      E-mail*
    </label>
    <input
      className="form-input"
      id="email"
      minLength={undefined}
      name="email"
      required={true}
      type="email"
    />
  </div>
  <div
    className="form-field"
  >
    <label
      className="form-label"
      htmlFor="phone"
    >
      Telefono
    </label>
    <input
      className="form-input"
      id="phone"
      minLength={undefined}
      name="phone"
      required={false}
      type="tel"
    />
  </div>
  <div
    className="form-field"
  >
    <label
      className="form-label"
      htmlFor="website"
    >
      Sito
    </label>
    <input
      className="form-input"
      id="website"
      minLength={undefined}
      name="website"
      required={false}
      type="text"
    />
  </div>
  <div
    className="form-field"
  >
    <span
      className="form-label"
    >
      Motivo della richiesta
      *
    </span>
    <fieldset
      className="form-input form-reason"
    >
      <span
        className="form-reason-option"
      >
        <input
          id="project"
          name="reason"
          required={true}
          type="radio"
          value="project"
        />
        <label
          htmlFor="project"
        >
          Elaborazione progetto
        </label>
      </span>
      <span
        className="form-reason-option"
      >
        <input
          id="call"
          name="reason"
          required={true}
          type="radio"
          value="call"
        />
        <label
          htmlFor="call"
        >
          Miglioria per gara d'appalto
        </label>
      </span>
      <span
        className="form-reason-option"
      >
        <input
          id="other"
          name="reason"
          required={true}
          type="radio"
          value="other"
        />
        <label
          htmlFor="other"
        >
          Altro
        </label>
      </span>
    </fieldset>
  </div>
  <label
    className="form-field"
  >
    <span
      className="form-label"
    >
      Eventuali informazioni aggiuntive
    </span>
    <textarea
      className="form-msg form-input"
      id="msg"
      name="msg"
    />
  </label>
  <button
    className="ctaBtn form-submit"
    type="submit"
  >
    Invia
  </button>
</form>
`;
